Jenkinsfile

```
@@ -11,7 +11,8 @@ buildPlugin(
   artifactCachingProxyEnabled: true,
   // Test Java 11 with minimum Jenkins version, Java 17 with a more recent version
   configurations: [
-    [platform: 'linux',   jdk: '11'], // Linux first for coverage report on ci.jenkins.io
+    [platform: 'linux',   jdk: '11', jenkins: '2.356'], // Linux first for coverage report on ci.jenkins.io
```

The plugin requires Jenkins 2.361.4 or newer.  Testing with Jenkins 2.356 won't work.
```suggestion
    [platform: 'linux',   jdk: '11'], // Linux first for coverage report on ci.jenkins.io
```

---

Why did ci build not run for jdk 17?

---

Because the `Jenkinsfile` is a trusted file and changes to it are only executed if you have merge permission to the repository.

Imagine the chaos that could be created by a malicious pull request submitter if their changes to Jenkinsfile were executed without review by a trusted maintainer.

---

I am very new to contributing to open source....In that case the PR I opened should not even trigger a CI build unless a trusted maintainer approved it...The change I made includes linux  - jdk 11, 17 and Windows...I see pipeline run for Linux -jdk 11 and windows but not linux - jdk 17 ...Exclusions/inclusions are controlled in some other place?

---

For sure...Thank you for explaining things.

---

No, the CI build is triggered, but your proposed change to the Jenkinsfile is ignored because you're not a maintainer.

I consider it overkill to test Java 17 on Linux and Java 17 on Windows.  If you are persuaded that you really need Linux with Java 17, then switch Windows to test with Java 11.  That way we're only running one Windows and one Linux, with one Java 11 and one Java 17.